awsx/ec2/vpc.ts

```diff
@@ -495,14 +495,16 @@ export class Vpc extends schema.Vpc<VpcData> {
       parsedSpecs !== undefined &&
       parsedSpecs.isExplicitLayout === false
     ) {
-      pulumi.log.warn(
-        `The default subnetStrategy will change from "Legacy" to "Auto" in the next major version. Please specify the subnetStrategy explicitly. The current subnet layout can be specified via "Auto" as:\n\n${JSON.stringify(
-          subnetLayout,
-          undefined,
-          2,
-        )}`,
-        this,
-      );
+      subnetLayout.apply(sl => {
```

Comment:
Good catch!
Is this `apply` guaranteed to execute/complete? What if we return the `subnetLayout` again in the apply and thread that through to the function's return statement to ensure it's executed?

Something like:
```// Only warn if they're using a custom, non-explicit layout and haven't specified a strategy.
    const verifiedSubnetLayout = subnetLayout.apply(sl => {
      if (
        args.subnetStrategy === undefined &&
        parsedSpecs !== undefined &&
        parsedSpecs.isExplicitLayout === false
      ) {
        pulumi.log.warn(
          `The default subnetStrategy will change from "Legacy" to "Auto" in the next major version. Please specify the subnetStrategy explicitly. The current subnet layout can be specified via "Auto" as:\n\n${JSON.stringify(
            sl,
            undefined,
            2,
          )}`,
          this,
        );
      }

      return sl;
    });
    
    return { subnetLayout: verifiedSubnetLayout, subnetSpecs };
```

Reply:
It won't execute until the Output is resolved, but then it will. I don't think that your change will make a difference in practice - but maybe it's a win in terms of readability and not provoking future engineers to ask this question again.

Reply:
Eh, I wouldn't be sure. Programs await all outstanding apply threads before exiting the program, but I'm not 100% sure the same is certain of code inside MLC contexts . So it might be possible this introduces a race. 

If it is the case however the best place to fix it would be the node SDK not the program here.

Reply:
In cases where it's' critical to await some effect and we're still not sure about the above, one trick is to enforce ordering by making the effect output an antecedent of some awaited output. 

```
foo(): Output<X> {
    effectOutput = something.apply(xValue => doSometingImportant(xValue))
    outputX = ...
    return effectOutput.apply(_ => outputX)
}
```